packages/dds/sequence/src/intervals/intervalUtils.ts

```diff
@@ -264,19 +264,47 @@ export const IntervalStickiness = {
 export type IntervalStickiness = (typeof IntervalStickiness)[keyof typeof IntervalStickiness];
 
 export function startReferenceSlidingPreference(
-	stickiness: IntervalStickiness,
+	startPos: number | "start" | "end" | undefined,
+	startSide: Side,
+	endPos: number | "start" | "end" | undefined,
+	endSide: Side,
 ): SlidingPreference {
+	const stickiness = computeStickinessFromSide(startPos, startSide, endPos, endSide);
 	// if any start stickiness, prefer sliding backwards
 	return (stickiness & IntervalStickiness.START) === 0
 		? SlidingPreference.FORWARD
 		: SlidingPreference.BACKWARD;
 }
 
 export function endReferenceSlidingPreference(
-	stickiness: IntervalStickiness,
+	startPos: number | "start" | "end" | undefined,
+	startSide: Side,
+	endPos: number | "start" | "end" | undefined,
+	endSide: Side,
 ): SlidingPreference {
+	const stickiness = computeStickinessFromSide(startPos, startSide, endPos, endSide);
+
 	// if any end stickiness, prefer sliding forwards
 	return (stickiness & IntervalStickiness.END) === 0
 		? SlidingPreference.BACKWARD
 		: SlidingPreference.FORWARD;
 }
+
+export function computeStickinessFromSide(
+	startPos: number | "start" | "end" | undefined,
+	startSide: Side,
+	endPos: number | "start" | "end" | undefined,
+	endSide: Side,
+): IntervalStickiness {
+	let stickiness: IntervalStickiness = IntervalStickiness.NONE;
+
+	if (startSide === Side.After || startPos === "start") {
```

Comment:
this is pre-existing so probably don't try to clean up in this PR, but the second clause here *should* be implied by the first one. You mentioned at standup that this is apparently not true, so there's probably some bugs lurking that need to be fixed, but after you do that, this should allow simplifying this method as well as the ones above. In fact, sliding preference for a given local reference should be a pure and invertible function of the `side` specified in the SequencePlace that the app author uses. `Side.Before` <==> `SlidingPreference.FORWARD` and `Side.After` <==> `SlidingPreference.BACKWARD`.